File: android/app/src/main/kotlin/store/swust/swustmeow/widgets/today_courses/TodayCoursesWidgetState.kt
package store.swust.swustmeow.widgets.today_courses

import store.swust.swustmeow.entities.SingleCourse

data class TodayCoursesWidgetState(
    val success: Boolean = false,
    val lastUpdateTimestamp: Long = 0,
    val todayCourses: List<SingleCourse>? = null,
    val weekNum: Int = 0
)

File: lib/widgets/today_courses/today_courses_widget_state.dart
import 'package:flutter/cupertino.dart';

import '../entities/single_course.dart';

class TodayCoursesWidgetState {
  final success = ValueNotifier(true);
  final lastUpdateTimestamp =
      ValueNotifier(DateTime.now().millisecondsSinceEpoch);
  final todayCourses = ValueNotifier<List<SingleCourse>?>(null);
  final weekNum = ValueNotifier<int>(0);

  void clear() {
    todayCourses.value = null;
    weekNum.value = 0;
  }
}
